src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -0,0 +1,118 @@
+package picard.illumina.parser;
+
+import picard.illumina.ExtractIlluminaBarcodes;
+import picard.illumina.ExtractIlluminaBarcodes.BarcodeMetric;
+import picard.illumina.parser.readers.AbstractIlluminaPositionFileReader;
+import picard.illumina.parser.readers.CbclReader;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static picard.illumina.ExtractIlluminaBarcodes.PerTileBarcodeExtractor.findBestBarcodeAndUpdateMetrics;
```

Comment:
Do we have a policy on static imports? I think in general it would be better to avoid them. In this case it looks like `PerTileBarcodeExtractor` should be promoted to a top-level class, to make it easier to work with.

My preference is to only allow them in test code, for using APIs like mockito that read better with static imports.